File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/value/factory/IntegerValueFactory.java
package ru.citeck.ecos.records3.record.op.atts.value.factory;

import ru.citeck.ecos.records3.record.op.atts.value.AttValue;

import java.util.Collections;
import java.util.List;

public class IntegerValueFactory implements AttValueFactory<Integer> {

    @Override
    public AttValue getValue(Integer value) {
        return new AttValue() {
            @Override
            public String getString() {
                return value.toString();
            }

            @Override
            public Double getDouble() {
                return Double.valueOf(value);
            }

            @Override
            public Boolean getBool() {
                return value != 0;
            }
        };
    }

    @Override
    public List<Class<? extends Integer>> getValueTypes() {
        return Collections.singletonList(Integer.class);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/atts/value/factory/IntegerValueFactory.kt
package ru.citeck.ecos.records3.record.atts.value.factory

import ru.citeck.ecos.records3.record.atts.value.AttValue

class IntegerValueFactory : AttValueFactory<Int> {

    override fun getValue(value: Int): AttValue {

        return object : AttValue {

            override fun asText(): String {
                return value.toString()
            }

            override fun asDouble(): Double {
                return java.lang.Double.valueOf(value.toDouble())
            }

            override fun asBoolean(): Boolean {
                return value != 0
            }

            override fun asRaw(): Any {
                return value
            }
        }
    }

    override fun getValueTypes() = listOf(Integer::class.java)
}
